src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -647,7 +647,7 @@ private void ProduceTagsSynchronously(TaggerContext<TTag> context)
                 {
                     foreach (var (latestBuffer, latestSpans) in newTagTrees)
                     {
-                        var snapshot = spansToTag.First(s => s.SnapshotSpan.Snapshot.TextBuffer == latestBuffer).SnapshotSpan.Snapshot;
+                        var snapshot = spansToTag.FirstOrDefault(s => s.SnapshotSpan.Snapshot.TextBuffer == latestBuffer).SnapshotSpan.Snapshot ?? latestBuffer.CurrentSnapshot;
```

Comment:
this is scary to me.  esp. without a comment.  When are we asking for spans to tag... but then none of those spans to tag match the buffer we care about...

Reply:
➡️ This is just building on #27933.